src/spikeinterface/sortingcomponents/motion_interpolation.py

```diff
@@ -106,53 +109,45 @@ def interpolate_motion_on_traces(
-    #      this would remove the need of _get_closest_ind and searchsorted
+    total_num_chans = channel_locations.shape[0]
 
-    # TODO delegate times to recording, at the moment this is 0 based
-    # regroup times by closet temporal_bins
-    bin_inds = _get_closest_ind(motion.temporal_bins_s[segment_index], times)
+    # -- determine the blocks of frames that will land in the same interpolation time bin
+    time_bins = interpolation_time_bin_centers_s
+    if time_bins is None:
+        time_bins = motion.temporal_bins_s[segment_index]
+    bin_s = time_bins[1] - time_bins
+    bins_start = time_bins[0] - 0.5 * bin_s
+    # nearest bin center for each frame?
+    bin_inds = (times - bins_start) // bin_s
+    # the time bins may not cover the whole set of times in the recording,
+    # so we need to clip these indices to the valid range
+    np.clip(bin_inds, 0, time_bins.size, out=bin_inds)
+
+    # -- what are the possibilities here anyway?
+    bins_here = np.arange(bin_inds[0], bin_inds[-1] + 1)
 
     # inperpolation kernel will be the same per temporal bin
-    for bin_ind in np.unique(bin_inds):
-
-        bin_time = motion.temporal_bins_s[segment_index][bin_ind]
-
+    interp_times = np.empty(total_num_chans)
+    current_start_index = 0
+    for bin_ind in bins_here:
+        bin_time = time_bins[bin_ind]
+        interp_times.fill(bin_time)
         channel_motions = motion.get_displacement_at_time_and_depth(
-            np.full(total_num_chans, bin_time),
+            interp_times,
             channel_locations[motion.dim],
-            segment_index=segment_index
+            segment_index=segment_index,
         )
         channel_locations_moved = channel_locations.copy()
         channel_locations_moved[:, motion.dim] += channel_motions
 
-        # # TODO use # TODO : add Motion.get_displacement_at_time_and_depth() instead
-
-        # # Step 1 : channel motion
-        # if spatial_bins.shape[0] == 1:
-        #     # rigid motion : same motion for all channels
-        #     channel_motions = motion[bin_ind, 0]
-        # else:
-        #     # non rigid : interpolation channel motion for this temporal bin
-        #     f = scipy.interpolate.interp1d(
-        #         spatial_bins, motion[bin_ind, :], kind="linear", axis=0, bounds_error=False, fill_value="extrapolate"
-        #     )
-        #     locs = channel_locations[:, direction]
-        #     channel_motions = f(locs)
-        # channel_locations_moved = channel_locations.copy()
-        # channel_locations_moved[:, direction] += channel_motions
-        # # channel_locations_moved[:, direction] -= channel_motions
-
         if channel_inds is not None:
             channel_locations_moved = channel_locations_moved[channel_inds]
 
         drift_kernel = get_spatial_interpolation_kernel(
             channel_locations,
             channel_locations_moved,
-            dtype="float32",
+            dtype=traces.dtype,
```

Comment:
dangerous no if the dtype is int16 ? The kriging is mostly between 0 and 1.

Reply:
Good point! revised the dtype handling now -- I was trying to support other floating dtypes than just float32, and hopefully now that's handled in a better way. I'm on a plane and the wifi blocks my ssh remote but will push tomorrow !